packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -132,6 +134,58 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
         }
     }
     
+
+    func onDelete(args: [String: Any], flutterResult: @escaping FlutterResult) {
+        do {
+            guard let modelName = args["modelName"] as? String else {
+                FlutterDataStoreErrorHandler.prepareError(
+                    flutterResult: flutterResult,
+                    msg: FlutterDataStoreErrorMessage.MALFORMED.rawValue,
+                    errorMap: ["MALFORMED_REQUEST": "modelName was not passed in the arguments." ])
+                return
+            }
+            guard let rawModel = args["model"] as? Dictionary<String, Any> else {
+                FlutterDataStoreErrorHandler.prepareError(
+                    flutterResult: flutterResult,
+                    msg: FlutterDataStoreErrorMessage.MALFORMED.rawValue,
+                    errorMap: ["MALFORMED_REQUEST": "model was not passed in the arguments." ])
+                return
+            }
+            guard let id = rawModel["id"] as? String else {
+                FlutterDataStoreErrorHandler.prepareError(
+                    flutterResult: flutterResult,
+                    msg: FlutterDataStoreErrorMessage.MALFORMED.rawValue,
+                    errorMap: ["MALFORMED_REQUEST": "model did not contain an id." ])
+                return
+            }
+
+            let modelData = SerializedModel(id: id, map: getJSONValue(rawModel))
+            
+            try bridge.onDelete(id: id,
+                              modelData: modelData,
+                              modelSchema: flutterModelRegistration.modelSchemas[modelName]!) { (result) in
```

Comment:
put a guard on the modelSchema to avoid a force cast

Reply:
Done.